tensorflow/compiler/xla/service/buffer_assignment.cc

```diff
@@ -83,7 +83,8 @@ std::vector<int64> ColorInterferenceGraph(
 
     // Find the color that is not yet assigned to the neighbors.
     int64 color = kColorUnassigned;
-    for (color = 0; color < available_colors.size(); ++color) {
+    const int64 end = available_colors.size();
+    for (color = 0; color < end; ++color) {
```

Comment:
Why was this extracted out instead of using the same pattern as the other loops?

Reply:
Was following the form that was already present.

`color` was initialized outside of loop so I initialize end outside of the loop as well.